### ISSUE TITLE
How to Install and Run Ultralytics YOLO11 for Real-Time Object Detection on Raspberry Pi 5

### ISSUE THREAD

**Issue by pornpra** [ultralytics/ultralytics#22391]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi everyone,

I’m trying to set up Ultralytics YOLO11 on my Raspberry Pi 5 to perform real-time object detection from an RTSP camera stream. Before starting, I’d like to confirm a few important details to make sure I install it correctly and avoid compatibility issues.

**System Setup Questions**

1. Operating System – Which Raspberry Pi OS is compatible with Ultralytics and PyTorch?

- Is it better to use 64-bit OS instead of the preinstalled 32-bit version for running YOLO11 efficiently?

2. Python Version – Which Python version is recommended?

- Can I use Python 3.13, or should I install an earlier version (e.g., 3.10 or 3.11) for better support with PyTorch and Ultralytics?

3. Dependencies Installation – What are the correct installation commands for Raspberry Pi 5?

- Example:

`sudo apt update && sudo apt upgrade -y`
`sudo apt install python3-pip`
`pip install torch torchvision torchaudio`
`pip install ultralytics`

- Are there any additional dependencies (like libopenblas-dev, opencv-python, etc.) required for successful installation?

4. Performance Tips – Any practical suggestions for improving performance on the Pi 5 (e.g., using swap space, hardware acceleration, lightweight models like yolo11n.pt, or exporting to ONNX/TensorRT)?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @pornpra, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started while an Ultralytics engineer reviews your question and will assist soon. We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For your Raspberry Pi 5 setup and real-time RTSP detection with YOLO11, please share any installation or runtime issues you encounter along with:
- Exact OS version and Python version
- Commands you ran and their full outputs
- Any error messages or logs
- Model and mode you’re trying (e.g., yolo11n, detection, export)
- Whether you’re using system OpenCV or pip-installed opencv-python
This will help us guide you more precisely ⚙️

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Most of the information can be found in the docs:

https://docs.ultralytics.com/guides/raspberry-pi/

1. You can use the latest
2. 64-bit maybe better
3. Python 3.12 works best
4. Check docs for this
5. Also in docs along with benchmarks
